main.go

```diff
@@ -100,6 +110,12 @@ func main() {
 			Usage:  "Private key used in client TLS auth",
 			Value:  "",
 		},
+		cli.StringFlag{
+			EnvVar: "MACHINE_SSH_CLIENT_TYPE",
+			Name:   "ssh-client-type",
+			Usage:  "Type of SSH client to use.  Options: gonative, shellout",
+			Value:  "",
```

Comment:
Honestly the option almost seems like it should be a one-way thing, if it's set then use native, if not then check if ssh bin is available or not.